File: Task/Munching-squares/Perl/munching-squares.pl
use GD;

my $img = new GD::Image(256, 256, 1);

for my $y(0..255) {
        for my $x(0..255) {
                my $color = $img->colorAllocate( abs(255 - $x - $y),  (255-$x) ^ $y , $x ^ (255-$y));
                $img->setPixel($x, $y, $color);
        }
}

print $img->png

File: Task/Munching-squares/Nim/munching-squares.nim
import random
import imageman

randomize()

# Build a color table.
var colors: array[256, ColorRGBU]
for color in colors.mitems:
  color = ColorRGBU [byte rand(255), byte rand(255), byte rand(255)]


var image = initImage[ColorRGBU](256, 256)

for i in 0..255:
  for j in 0..255:
    image[i, j] = colors[i xor j]

image.savePNG("munching_squares.png")
